src/bug-filing/common/http-query-builder.ts

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { size, values } from 'lodash';
+import { DictionaryStringTo } from '../../types/common-types';
+
+export type QueryParam = {
```

Comment:
It feels redundant to have name duplicated in the dictionary key and value:
```
{
    n1: { name: n1, value: v1 },
    n2: { name: n2, value: v2 },
}
```
Seems like it would be simpler to have
```
{
    n1: v1,
    n2: v2,
}
```
Key to doing this is using lodash's `toPairs` rather than `values` at line 39.

Reply:
Using such a dictionary and `toPairs` render me to this:

```
const queryParameters = toPairs(this.parameters)
    .map(pair => '${encodeURIComponent(pair[0])}=${encodeURIComponent(pair[1])}')
    .join('&');
```

where accessing the name and the value doesn't look too elegant.

The other option is to use `Object.keys(this.parameters)` and use the resulting array with a `forEach` or `for` loop. `forEach` will land on the same place as `toPairs` while `for` you have one more line of code to access the current element in the array using the current key.

I'll prefer to keep this little redundancy (not to bad really) so I can have the more verbose version of the `map` usage.